consensus/abci_consensus_connection.go

```diff
@@ -0,0 +1,220 @@
+	}
+
+	err = abci.ResetBlockState()
+	if err != nil {
+		return nil, err
+	}
+
+	return abci, nil
+}
+
+// InitChain is called upon genesis.
+//
+// Can be used to define validators set and consensus params on the application side.
+//
+// Response:
+//     - If `ResponseInitChain.Validators` is empty, the initial validator set will be the `RequestInitChain.Validators`
+//
+func (abci *TendermintABCI) InitChain(req tmtAbciTypes.RequestInitChain) tmtAbciTypes.ResponseInitChain {
+	abci.logger.Debug("Initializing chain", "chain_id", req.ChainId)
+
+	return tmtAbciTypes.ResponseInitChain{}
+}
+
+// BeginBlock signals the beginning of a new block.
+//
+// Flow:
+//		1. BeginBlock <- ******
+//		2. CheckTx
+//	    3. ExecuteTx
+//		4. EndBlock
+//		5. Commit
+//		6. CheckTx (clean mempool from TXs not included in committed block)
+//
+// The header contains the height, timestamp, and more - it exactly matches the Tendermint block header.
+// The LastCommitInfo and ByzantineValidators can be used to determine rewards and punishments for the validators.
+//
+// Response:
+//		- Optional Key-Value tags for filtering and indexing
+//
+func (abci *TendermintABCI) BeginBlock(req tmtAbciTypes.RequestBeginBlock) tmtAbciTypes.ResponseBeginBlock {
+	abci.logger.Debug("Beginning new block", "hash", req.Hash)
+	abci.db.UpdateHeaderWithTimeInfo(&req.Header)
+
+	return tmtAbciTypes.ResponseBeginBlock{}
+}
+
+// ExecuteTx executes the transaction against Ethereum block's work state.
+//
+// Flow:
+//		1. BeginBlock
+//		2. CheckTx
+//	    3. ExecuteTx <- ******
+//		4. EndBlock
+//		5. Commit
+//		6. CheckTx (clean mempool from TXs not included in committed block)
+//
+// Tendermint runs CheckTx and ExecuteTx concurrently with each other,
+// though on distinct ABCI connections - the mempool connection and the consensus connection, respectively.
+//
+// Response:
+//		- If the transaction is valid, returns CodeType.OK
+//		- Keys and values in Tags must be UTF-8 encoded strings
+// 		  E.g: ("account.owner": "Bob", "balance": "100.0", "time": "2018-01-02T12:30:00Z")
+//
+func (abci *TendermintABCI) DeliverTx(txBytes []byte) tmtAbciTypes.ResponseDeliverTx {
+	tx, err := decodeRLP(txBytes)
+	if err != nil {
+		abci.logger.Info("Received invalid transaction", "hash", tx, "err", err)
+
+		return tmtAbciTypes.ResponseDeliverTx{Code: uint32(abciTypes.ErrEncodingError.Code), Log: err.Error()}
+	}
+
+	abci.logger.Info("Delivering TX", "hash", tx.Hash().String(), "nonce", tx.Nonce(), "cost", tx.Cost(), "gas", tx.Gas(), "gas_price", tx.GasPrice())
+
+	res := abci.db.ExecuteTx(tx)
+	if res.IsErr() {
+		abci.logger.Error("Error delivering TX to DB", "hash", tx.Hash().String(), "err", res.Log)
+
+		return res
+	}
+
+	abci.logger.Info("TX delivered.", "tx", tx.Hash().String())
+
+	return tmtAbciTypes.ResponseDeliverTx{Code: tmtAbciTypes.CodeTypeOK}
+}
+
+// EndBlock signals the end of a block.
+//
+// Flow:
```

Comment:
I think having these flow description above every method, it does the documentation quiet tedious to read